python/ray/ml/predictors/integrations/tensorflow/tensorflow_predictor.py

```diff
@@ -0,0 +1,123 @@
+from typing import Optional, Union, List, Type
+
+import tensorflow as tf
+
+from ray.ml.predictor import Predictor, DataBatchType
+from ray.ml.preprocessor import Preprocessor
+from ray.ml.checkpoint import Checkpoint
+
+# TensorFlow model objects cannot be pickled, therefore we use
+# a callable that returns the model, instead of a model object
+# itself.
+
+
+class TensorflowPredictor(Predictor):
+    """A predictor for TensorFlow models.
+
+    Args:
+        model_definition: A callable that returns a TensorFlow Keras model
+            to use for predictions.
+        preprocessor: A preprocessor used to transform data batches prior
+            to prediction.
+        model_weights: List of weights to use for the model.
+    """
+
+    def __init__(
+        self,
+        model_definition: Type[tf.keras.Model],
+        preprocessor: Preprocessor,
+        model_weights: Optional[list] = None,
+    ):
+        raise NotImplementedError
+
+    @classmethod
+    def from_checkpoint(
+        cls, checkpoint: Checkpoint, model_definition: Type[tf.keras.Model]
+    ) -> "TensorflowPredictor":
+        """Instantiate the predictor from a Checkpoint.
+
+        The checkpoint is expected to be a result of ``TorchTrainer``.
+
+        Args:
+            checkpoint: The checkpoint to load the model and
+                preprocessor from. It is expected to be from the result of a
+                ``TorchTrainer`` run.
+            model_definition: A callable that returns a TensorFlow Keras model
+                to use. Model weights will be loaded from the checkpoint.
+        """
+        raise NotImplementedError
+
+    def predict(
+        self,
+        data: DataBatchType,
+        column_names: Optional[List[str]] = None,
```

Comment:
I originally had this as well, but no longer necessary I feel. If columns don’t have names you can still specify them with int indices.